reVX/plexos/plexos_plants.py

```diff
@@ -0,0 +1,1006 @@
+        """
+        Create Supply Curve Point objects, used for filling plants
+
+        Parameters
+        ----------
+        max_workers : int, optional
+            Number of workers to use for point and plant creation, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+        points_per_worker : int, optional
+            Number of points to create on each worker, by default 400
+
+        Returns
+        -------
+        sc_points : list
+            List of Supply Curve Point objects
+        """
+        sc_points = SupplyCurvePoints(self.sc_table, self._cf_fpath,
+                                      max_workers=max_workers,
+                                      points_per_worker=points_per_worker,
+                                      offshore=self._offshore)
+
+        return sc_points
+
+    def _allocate_sc_gids(self, sc_gids, dists, bus_dists):
+        """
+        Allocate capacity from supply curve points to plants
+
+        Parameters
+        ----------
+        sc_gids : list
+            List of supply curve point gids to allocate capacity from
+        dists : list
+            List of distances from plants to sc_gids
+        bus_dists : list
+            List of distances from bus associated with plants to sc_gids
+        """
+        unique_gids, plant_gids = np.unique(sc_gids, return_inverse=True)
+        for i, sc_gid in enumerate(unique_gids):
+            sc_gid = int(sc_gid)
+            if self.sc_points.check_sc_gid(sc_gid):
+                plant_ids = np.where(plant_gids == i)[0]
+                if len(plant_ids) > 1:
+                    sc_dists = dists[plant_ids]
+                    if len(sc_dists) != len(np.unique(sc_dists)):
+                        idxs = np.argsort(bus_dists[plant_ids])
+                    else:
+                        idxs = np.argsort(sc_dists)
+
+                    plant_ids = plant_ids[idxs]
+
+                for plant_id in plant_ids:
+                    capacity = self.plant_capacity[plant_id]
+                    if (capacity > 0) and self.sc_points.check_sc_gid(sc_gid):
+                        sc_point, sc_capacity = \
+                            self.sc_points.get_capacity(sc_gid, capacity)
+                        if sc_capacity:
+                            plant = self[plant_id]
+                            if plant is None:
+                                plant = [sc_point]
+                            else:
+                                plant.append(sc_point)
+
+                            self[plant_id] = plant
+                            self._capacity[plant_id] -= sc_capacity
+                            logger.debug('Allocating {:.1f}MW to plant {} from'
+                                         ' sc_gid {}'.format(sc_capacity,
+                                                             plant_id,
+                                                             sc_gid))
```

Comment:
I find this wildly confusing.... Is plant a PlexosPlant object? Why are you appending an SC point to the plant? I think in reality the plant is a list of SC points associated with each plant, but then why is it called plant? Maybe we should break out the "plant" into a standalone object.